### PR TITLE
Add test for access count in Hive metastore

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Add test for access count in Hive metastore

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+
+        assertMetastoreInvocations("CALL system.drop_stats('test_schema', 'drop_stats_partition')",
+                ImmutableMultiset.builder()
+                        .addCopies(GET_TABLE, 2)
+                        .add(GET_PARTITIONS)
+                        .addCopies(UPDATE_PARTITION_STATISTICS, 2)
+                        .build());
+    }
+
+    private void assertMetastoreInvocations(@Language("SQL") String query, Multiset<?> expectedInvocations)
```
[the remark this answers is not in the document]
Comment (ebyhr) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java:275]: Rebased to use `CountingAccessHiveMetastoreUtil.assertMetastoreInvocations` method.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    {
+        assertUpdate("CREATE TABLE test_select_from (id VARCHAR, age INT)");
+
+        assertMetastoreInvocations("SELECT * FROM test_select_from",
+                ImmutableMultiset.builder()
+                        .add(GET_TABLE)
+                        .addCopies(GET_TABLE_STATISTICS, 1)
+                        .build());
+    }
+
```
Comment (findinpath) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java:104]: Could you please add a test for selecting from a partitioned table?

Comment (ebyhr) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java:104]: Added.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+    @Test
+    public void testSelectFromView()
+    {
+        assertUpdate("CREATE TABLE test_select_view_table (id VARCHAR, age INT)");
+        assertUpdate("CREATE VIEW test_select_view_view AS SELECT id, age FROM test_select_view_table");
+
+        assertMetastoreInvocations("SELECT * FROM test_select_view_view",
+                ImmutableMultiset.builder()
+                        .addCopies(GET_TABLE, 2)
+                        .addCopies(GET_TABLE_STATISTICS, 1)
```
Comment (findinpath) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java:154]:
```suggestion
                        .add(GET_TABLE_STATISTICS)
```

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+    @Test
+    public void testSelect()
+    {
+        assertUpdate("CREATE TABLE test_select_from (id VARCHAR, age INT)");
+
+        assertMetastoreInvocations("SELECT * FROM test_select_from",
+                ImmutableMultiset.builder()
+                        .add(GET_TABLE)
+                        .addCopies(GET_TABLE_STATISTICS, 1)
```
Comment (findinpath) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java:101]:
```suggestion
                        .add(GET_TABLE_STATISTICS)
```

**Review by findinpath — APPROVED** [review]

**Review by findinpath — APPROVED** [review]

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/CountingAccessHiveMetastore.java [hunk 5]
```diff
@@ -208,19 +216,22 @@ public Optional<List<String>> getPartitionNamesByFilter(String databaseName,
             List<String> columnNames,
             TupleDomain<String> partitionKeysFilter)
     {
-        throw new UnsupportedOperationException();
+        methodInvocations.add(Methods.GET_PARTITIONS);
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/CountingAccessHiveMetastore.java:219]: GET_PARTITION_NAMES_BY_FILTER

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/CountingAccessHiveMetastore.java [hunk 6]
```diff
@@ -208,19 +216,22 @@ public Optional<List<String>> getPartitionNamesByFilter(String databaseName,
...
+        return delegate.getPartitionNamesByFilter(databaseName, tableName, columnNames, partitionKeysFilter);
     }
 
     @Override
     public Map<String, Optional<Partition>> getPartitionsByNames(Table table, List<String> partitionNames)
     {
-        throw new UnsupportedOperationException();
+        methodInvocations.add(Methods.GET_PARTITIONS);
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/CountingAccessHiveMetastore.java:226]: GET_PARTITIONS_BY_NAMES

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+        queryRunner.createCatalog("hive", "hive", ImmutableMap.of());
+
+        queryRunner.execute("CREATE SCHEMA test_schema");
+        return queryRunner;
+    }
+
+    @Test
+    public void testCreateTable()
+    {
+        assertMetastoreInvocations("CREATE TABLE test_create (id VARCHAR, age INT)",
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java:72]: nit

```suggestion
        assertMetastoreInvocations("CREATE TABLE test_create(id VARCHAR, age INT)",
```

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java [hunk 8]
```diff
@@ -0,0 +1,307 @@
...
+                        .add(CREATE_TABLE)
+                        .add(GET_TABLE)
+                        .add(UPDATE_TABLE_STATISTICS)
+                        .build());
+    }
+
+    @Test
+    public void testSelect()
+    {
+        assertUpdate("CREATE TABLE test_select_from (id VARCHAR, age INT)");
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreAccessOperations.java:96]: nit

```suggestion
        assertUpdate("CREATE TABLE test_select_from(id VARCHAR, age INT)");
```

**Review by findepi — APPROVED** [review]

**Pull request merged**